### PR TITLE
Deprecate `agg_key_funcs`, `agg_default_func`, and `update_agg_funcs` from `LightningLoggerBase`

### PR DESCRIPTION
**Pull request by akashkw**
## What does this PR do?
Deprecates `agg_key_funcs`, `agg_default_func` and `update_agg_funcs` from `LightningLoggerBase`. 

Part of #11235 
Part of #9145 

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by edward-io** [comment]
Code looks good. For the deprecation message, can we add more information on what alternatives users could use?

**Review by ananthsub — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -391,6 +391,12 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Deprecated `pytorch_lightning.utilities.warnings.LightningDeprecationWarning` in favor of `pytorch_lightning.utilities.rank_zero.LightningDeprecationWarning`
 
 
+- Deprecated `agg_key_funcs` and `agg_default_func` parameters from `LightningLoggerBase` ([#11871](https://github.com/PyTorchLightning/pytorch-lightning/pull/11871))
+
+
+- Deprecated `LightningLoggerBase.update_agg_funcs` because `agg_key_funcs` and `agg_default_func` are deprecated ([#11871](https://github.com/PyTorchLightning/pytorch-lightning/pull/11871))
```
Comment (daniellepintz) [inline CHANGELOG.md:397]:
```suggestion
- Deprecated `LightningLoggerBase.update_agg_funcs` ([#11871](https://github.com/PyTorchLightning/pytorch-lightning/pull/11871))
```

**Comment by daniellepintz** [comment]
Can you also deprecate these in this PR, or were you planning to do this in another PR? 
https://github.com/PyTorchLightning/pytorch-lightning/blob/79c4e5de60685dbc895641b0139ffc6180d069aa/pytorch_lightning/loggers/base.py#L102-L146

For these methods, you can put the dep. message in the docstring, but no need to send out a dep warning

**Comment by akashkw** [comment]
> Code looks good. For the deprecation message, can we add more information on what alternatives users could use?

@edward-io  Since the aggregation logic is in the training loop now, I'm not sure how (or if) a user can provide custom aggregation functions. Do you know what alternatives there might be?

**Comment by akashkw** [comment]
> Can you also deprecate these in this PR, or were you planning to do this in another PR?

I can do that in this PR.

**Comment by edward-io** [comment]
> > Code looks good. For the deprecation message, can we add more information on what alternatives users could use?
> 
> @edward-io Since the aggregation logic is in the training loop now, I'm not sure how (or if) a user can provide custom aggregation functions. Do you know what alternatives there might be?

Mentioning this in the deprecation message would be sufficient IMO, but this is just a nit. :)

**Comment by daniellepintz** [comment]
Since the aggregation logic is currently a no-op, I don't think it's necessary to mention any alternatives to users, since we are not really removing any functionality. I suppose we could mention the fact that the agg logic is a no-op, but not sure if we  want to say that to users.

**Review by carmocca — APPROVED** [review]

**Comment by daniellepintz** [comment]
The CI error flaky, could someone please rerun the CI and enable auto-merge?
